### PR TITLE
[CU-860rj8ark]: Add realtime database

### PR DESCRIPTION
**Pull request by FlavianEng**
>__Warning__
> Environment variables have been updated. Look in `.env.sample` for more details.

>__Note__
> This PR makes some huge changes. We are no longer using our custom postgres docker for the application database but are now using supabase. As a result, in `bin/app` you'll now find an `open local supabase` command, giving you easy access to database data and more!

## Changelog:  
- Rework env files
- Setup supabase
- Rework `bin/app`
- Create default setttings table
- Connect realtime database

## Checklist before requesting a review
-   [x] I have performed a self-review of my code
-   [x] My pull request title respect the naming convention: [**_CU-{ClickUp Id}_**] **_{summary}_**

### CODE REVIEW

**Comment by FlavianEng** [comment]
Task linked: [CU-860rj8ark US: As a developer, I'd like to change default parameters in realtime](https://app.clickup.com/t/860rj8ark)

File: src/engine/game-presets.const.ts [hunk 1]
```diff
@@ -21,3 +19,16 @@ export const demoGame: GamePresetType = {
...
   name: PresetNameEnum.DEMO,
 };
+
+export function getGamePreset(presetName: PresetNameEnum) {
+  switch (presetName) {
+    case PresetNameEnum.DEMO:
+      return demoGame;
+
+    case PresetNameEnum.CLASSIC:
+      return classicGame;
```
Comment (MASBERNATQ) [inline src/engine/game-presets.const.ts:29]: Pas nécessaire comme tu l'as déjà en defaut de ton switch. Mais si tu souhaites vraiment le voir visuellement tu peux faire ceci :

```js
 case PresetNameEnum.CLASSIC:
 default:
      return classicGame;
```

File: src/shared/database.ts [hunk 2]
```diff
@@ -1,24 +1,22 @@
...
       break;
 
+    case EnvironmentEnum.TEST:
+      databaseUrl = process.env.DATABASE_TEST_URL;
+      break;
+
     default:
-      databaseUrl = `${process.env.DATABASE_URL}${databaseName}`;
+      databaseUrl = process.env.DATABASE_URL;
       break;
```
Comment (MASBERNATQ) [inline src/shared/database.ts:20]: Même commentaire un peu plus haut pour simplifier ton switch

File: src/engine/game-presets.const.ts (hunk 1 above)
Comment (FlavianEng) [reply src/engine/game-presets.const.ts:29]: Yes je sais que c'est pas nécessaire mais on en aura probablement plus dans le futur don je me suis dit queee ... bref okay je vais modifier

**Pull request merged**